vowpalwabbit/feature_group.cc

```diff
@@ -84,6 +94,12 @@ void features::concat(const features& other)
 
   if (!other.space_names.empty())
   { space_names.insert(space_names.end(), other.space_names.begin(), other.space_names.end()); }
+
+  for (const auto& ns_extent : other.namespace_extents)
+  {
+    namespace_extents.emplace_back(
+        ns_extent.begin_index + extent_offset, ns_extent.end_index + extent_offset, ns_extent.hash);
+  }
 }
```

Comment:
Could add logic here to combine last extent of first group with first extent of last group if they share the same hash

Reply:
Yeah youre right they should merge here if needed